krate/src/main/kotlin/hu/autsoft/krate/validated/exception/KrateValidationException.kt

```diff
@@ -0,0 +1,3 @@
+package hu.autsoft.krate.validated.exception
+
+class KrateValidationException(message: String) : Exception(message)
```

Comment:
This could be an `IllegalArgumentException` specifically instead of a generic `Exception`. And perhaps there isn't a need for this custom exception type, the `IllegalArgumentException` itself could just be thrown from the delegate directly.

Reply:
I like this point a lot. Good call out.